### ISSUE TITLE
Another crash in numpy.subtract (migrated from Trac #1896)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3444]
Original ticket http://projects.scipy.org/numpy/ticket/1896
Reported 2011-07-06 by atmention:r-owen, assigned to unknown.

I've seen this crash on MacOS X 10.6 using python.org Python 2.7.2 (32-bit backward compatible version) and three flavors of numpy:
- numpy 1.6.0 standard Mac binary
- numpy 1.6.0 built from source
- numpy 1.6.1rc2 built from source

I have attached a data file and script. Text in the script shows how to run it. You can choose to use Tkinter or not, but I find the bug shows up more reliably if you use Tkinter (the configure event triggers numpy.subtract).

**Comment by thouis** [comment]
Attachment in Trac by atmention:r-owen, 2011-07-06: [dataArr.zip](http://projects.scipy.org/numpy/attachment/ticket/1896/dataArr.zip)

**Comment by thouis** [comment]
Attachment in Trac by atmention:r-owen, 2011-07-06: [testmem.py](http://projects.scipy.org/numpy/attachment/ticket/1896/testmem.py)

**Comment by thouis** [comment]
Comment in Trac by atmention:cgohlke, 2011-07-06

Here's a reduced code that crashes on Windows with numpy 1.6 but works correctly with numpy 1.5:

```
import numpy as np
a = np.zeros(65536)
b = np.zeros(a.shape, dtype=np.float32)
np.subtract(a, 0, b)
```

np.add() has the same problem. Using dtype=np.float64 works.

**Comment by thouis** [comment]
Comment in Trac by atmention:cgohlke, 2011-07-06

**Comment by thouis** [comment]
Comment in Trac by atmention:r-owen, 2011-07-06

I can confirm that using dataArr = numpy.zeros(65536, dtype=float) does produce the problem on my system using the test script I originally attached. Thus you can ignore the data file.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2011-07-06

I see this also on master, 64 bit linux.

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2011-07-06

I don't have time to look at this the coming week, but I'll wait with the final 1.6.1 release till this is diagnosed.

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-07-06

I've made a pull request which should fix this:

https://github.com/numpy/numpy/pull/103

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2011-07-06

Do you ever sleep:)

I'll see about an RC3 tomorrow then.

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2011-07-11

Fixed in 1.6.1rc3.  Thanks all.
